src/Event/AfterInstantiate.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry\Event;
+
+use Zenstruck\Foundry\Factory;
+use Zenstruck\Foundry\ObjectFactory;
+
+/**
+ * @author Nicolas PHILIPPE <nikophil@gmail.com>
+ *
+ * @phpstan-import-type Parameters from Factory
+ */
+final class AfterInstantiate
```

Comment:
I hesitated to use `@template T` here, but this would force users to use it in listeners, and they will be tempted to do something like this:
```php
final class AddressEventListener
{
    /** @param BeforeInstantiate<Address> */
    #[AsEventListener]
    public function beforeInstantiate(BeforeInstantiate $event): void
    {
        if (Address::class !== $event->objectClass) {
            return;
        }

        // ...
    }
```
but phpstan will break on the if...

Reply:
Yeah, let's leave off for now